common/git-hooks/pre-commit

```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+# Called by "git commit" with no arguments.  The hook should
+# exit with non-zero status after issuing an appropriate message if
+# it wants to stop the commit.
+
+COMMAND=common/temp/node_modules/.bin/precise-commits
+
+echo --------------------------------------------
+echo Starting Git hook: pre-commit
+
+if [ -f $COMMAND ]; then
+  echo Invoking $COMMAND
+  $COMMAND
+else
+  echo Command not installed: $COMMAND
```

Comment:
Just wondering - should we actually exit with a non zero status here to block the commit or are we okay with allowing the commit to happen even if we were unable to run the prettier on it?

Reply:
I do not see why we might want to block the commit, do you have a reason in mind? Ideally, this case should never happen if `rush` is doing its job. However if for whatever reason `rush` state was corrupted and the binaries are missing, I doubt that aborting the commit will accomplish anything.

Reply:
I guess it's about how strong we are trying to push the enforcement of this behavior.  Blocking the commit would make it harder to end up checking in code in cases where your environment is horked, but maybe we think that's a rare enough occurrence and that even if the code gets checked in without having been run through prettier it's not the end of the world.

Happy with whatever call you make, was just trying to get a sense for how strict we were trying to be here.

Reply:
Yeah, sorry I did not make a good write-up about my approach earlier. I wrote the assumptions I work with in a [comment](https://github.com/Azure/azure-sdk-for-js/pull/10219#issuecomment-663737624) below.